app/utils/file/file_picker/index.ts

```diff
@@ -0,0 +1,331 @@
+                            id: 'mobile.permission_denied_dismiss',
+                            defaultMessage: "Don't Allow",
+                        }),
+                    },
+                ]);
+                return false;
+            }
+            default: return true;
+        }
+        return false;
+    };
+
+    private hasStoragePermission = async () => {
+        if (Platform.OS === 'ios') {
+            return true;
+        }
+
+        const storagePermission = Permissions.PERMISSIONS.ANDROID.READ_EXTERNAL_STORAGE;
+        let permissionRequest;
+        const hasPermissionToStorage = await Permissions.check(storagePermission);
+        switch (hasPermissionToStorage) {
+            case Permissions.RESULTS.DENIED:
+                permissionRequest = await Permissions.request(storagePermission);
+                if (permissionRequest !== Permissions.RESULTS.GRANTED) {
+                    return false;
+                }
+                break;
+            case Permissions.RESULTS.BLOCKED: {
+                const {title, text} = this.getPermissionDeniedMessage();
+
+                Alert.alert(title, text, [
+                    {
+                        text: this.intl.formatMessage({
+                            id: 'mobile.permission_denied_dismiss',
+                            defaultMessage: "Don't Allow",
+                        }),
+                    },
+                    {
+                        text: this.intl.formatMessage({
+                            id: 'mobile.permission_denied_retry',
+                            defaultMessage: 'Settings',
+                        }),
+                        onPress: () => AndroidOpenSettings.appDetailsSettings(),
+                    },
+                ]);
+                return false;
+            }
+            default: return true;
+        }
+
+        return false;
+    };
+
+    attachFileFromCamera = async (customOptions?: CameraOptions) => {
+        let options = customOptions;
+        if (!options) {
+            options = {
+                quality: 0.8,
+                videoQuality: 'high',
+                mediaType: 'photo',
+                saveToPhotos: true,
+            };
+        }
+
+        const hasCameraPermission = await this.hasPhotoPermission('camera');
+        if (hasCameraPermission) {
+            launchCamera(options, async (response: ImagePickerResponse) => {
+                StatusBar.setHidden(false);
+
+                if (response.errorCode || response.didCancel) {
+                    return;
+                }
+
+                const files = await this.getFilesFromResponse(response);
+                await this.prepareFileUpload(files);
+            });
+        }
+    };
+
+    attachFileFromFiles = async (browseFileType?: string) => {
+        const hasPermission = await this.hasStoragePermission();
+        const fileType = browseFileType ?? Platform.select({ios: 'public.item', android: '*/*'})!;
```

Comment:
Since all the platforms we hvae are ios or android, on Platform.select we can use one, and then default, so it returns always a defined type.

Reply:
Can you please clarify?

Reply:
what he means is that `Platform.select({...})` could return `undefined` if it does not find the platform thus TypeScript will force you to check for if the variable is defined.

Now `Platform.select({...})` accepts a `default` property, so for example if you do 

```suggestion
        const fileType = browseFileType ?? Platform.select({ios: 'public.item', default: '*/*'});
```

there won't be a need for the `!` bang operator